apps/docs/spec/supabase_js_v2.yml

```diff
@@ -237,6 +237,45 @@ functions:
         - When both **Confirm email** and **Confirm phone** (even when phone provider is disabled) are enabled in [your project](/dashboard/project/_/auth/providers), an obfuscated/fake user object is returned.
         - When either **Confirm email** or **Confirm phone** (even when phone provider is disabled) is disabled, the error message, `User already registered` is returned.
       - To fetch the currently logged-in user, refer to [`getUser()`](/docs/reference/javascript/auth-getuser).
+    overwriteParams:
+      - name: credentials 
+        type: SignUpWithPasswordCredentials
+        subContent:
+          - name: email
+            isOptional: true
+            type: string
+            description: One of `email` or `phone` must be provided.
+          - name: phone
+            isOptional: true
+            type: string
+            description: One of `email` or `phone` must be provided.
+          - name: password
+            type: string
+          - name: options
+            isOptional: true
+            type: object
+            subContent:
+              - name: emailRedirectTo
+                isOptional: true
+                type: string
+                description: >
+                  Only for email signups.
+                  The redirect URL embedded in the email link.
+                  Must be a configured redirect URL for your Supabase instance.
+              - name: data
+                isOptional: true
+                type: object
+                description: >
+                  A custom data object to store additional user metadata.
+              - name: captchaToken
+                isOptional: true
+                type: string
+              - name: channel
+                isOptional: true
+                type: sms | whatsapp
+                description: >
+                  The channel to use for sending messages.
+                  Only for phone signups.
```

Comment:
Safe to say this can't be generated upstream from `auth-js`?

Reply:
They aren't currently generated (the TS parsing ignores union types). I have a PR to improve this but it needs a few more iterations to clean it up, and as a quick fix doing it manually was fastest